src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -0,0 +1,277 @@
+
+/**
+ * A RoadNameSpellingConsistencyCheckWalker can be used to collect all edges that have NameTag
+ * values that are at an edit distance of 1 from the starting edge's {@link NameTag} value.
+ * Collected edges are within some linear search area (configurable).
+ *
+ * @author seancoulter
+ */
+class RoadNameSpellingConsistencyCheckWalker extends EdgeWalker
+{
+
+    /**
+     * Holds the unicode representation of CJK (Chinese, Japanese, Korean) Unified Ideograph numbers
+     * (e.g. 四) These numbers are not contained in {Nd} so we create separate storage for them
+     */
+    enum CJKNumbers
+    {
+        CJK_ZERO(0x3007),
+        CJK_ONE(0x4E00),
+        CJK_TWO(0x4E8C),
+        CJK_THREE(0x4E09),
+        CJK_FOUR(0x56DB),
+        CJK_FIVE(0x4E94),
+        CJK_SIX(0x516D),
+        CJK_SEVEN(0x4E03),
+        CJK_EIGHT(0x516B),
+        CJK_NINE(0x4E5D),
+        CJK_TEN(0x5341),
+        CJK_TWENTY(0x5EFF),
+        CJK_THIRTY(0x5345);
+
+        private final int value;
+
+        CJKNumbers(final int value)
+        {
+            this.value = value;
+        }
+
+        public int getValue()
+        {
+            return this.value;
+        }
+    }
+
+    // Matches identifiers sometimes found in road names. E.g. the 'A' in Road A, the "12c" in 12c
+    // Street, and the "Y6" in Y6 Drive.
+    // Identifiers are defined to be any space-delimited string that either contains at least one
+    // digit OR contains a single character which may be preceded or followed by a single
+    // punctuation character.
+    // We consider names with different identifiers to be from different roads, and as such we don't
+    // flag for spelling inconsistencies between identifiers.
+    private static final String ALPHANUMERIC_IDENTIFIER_STRING_REGEX = ".*\\p{Nd}+.*";
+    private static final String CHARACTER_IDENTIFIER_STRING_REGEX = "\\p{P}.\\p{P}|.\\p{P}|\\p{P}.|^.$";
+
+    private static final String WHITESPACE_REGEX = "\\s+";
+
+    /**
+     * Evaluate the {@link NameTag}s of the startingEdge and an incomingEdge to see if their
+     * spellings are inconsistent with one another. NOSONAR because startingEdges are guaranteed to
+     * have a name by way of validCheckForObject.
+     *
+     * @param startEdge
+     *            the edge from which the search started
+     * @return true if incomingEdge's name exists and is at an edit distance of 1 from the start
+     *         edge's name; false otherwise
+     */
+    @SuppressWarnings("squid:S3655")
+    static Predicate<Edge> isEdgeWithInconsistentSpelling(final Edge startEdge)
+    {
+        return incomingEdge ->
+        {
+            if (!incomingEdge.getName().isPresent())
+            {
+                return false;
+            }
+            final String startEdgeName = startEdge.getName().get();
+            final String incomingEdgeName = incomingEdge.getName().get();
+            return checkBeforeEditDistance(incomingEdgeName, startEdgeName);
```

Comment:
As a future improvement, I would be curious whether there are any cases where first doing Unicode normalization on the strings would change the result. For example, if there are cases where an accented character is decomposed in the name of one Way, and composed on another, such that the edit distance looks like two, when two equivalent strings would only have an edit distance of one from each other.

Reply:
Interesting. I could see something like this being used if we wanted to focus on spelling inconsistencies related to character numbers between the two strings as opposed to accent mixup.

Reply:
not completely relevant but as another future improvement if we want to handle more complex cases, we can also consider leveraging some libraries like Lucene. e.g. https://lucene.apache.org/core/7_3_1/suggest/org/apache/lucene/search/spell/LevenshteinDistance.html